.github/workflows/build_workflow.yml

```diff
@@ -10,6 +10,24 @@ on:
   workflow_dispatch:
 
 jobs:
+  pre-commit-hooks:
```

Comment:
`pre-commit-hooks` should be below `check-jobs-to-skip` for semantics.

For the first two lines under the title of the job, you should also include
```
    needs: check-jobs-to-skip
    if: ${{ needs.check-jobs-to-skip.outputs.should_skip != 'true' }}
```

For reference: 
https://github.com/E3SM-Project/e3sm_diags/blob/master/.github/workflows/build_workflow.yml#L25-L26